internal/csi-addons/rbd/volumegroup.go

```diff
@@ -343,17 +343,40 @@ func (vs *VolumeGroupServer) ModifyVolumeGroupMembership(
 		}
 	}
 
-	// add the new volumes to the group
-	for _, id := range toAdd {
-		vol, getErr := mgr.GetVolumeByID(ctx, id)
-		if getErr != nil {
+	// resolve all volumes
+	volumes := make([]types.Volume, len(toAdd))
+	for i, id := range toAdd {
+		var vol types.Volume
+		vol, err = mgr.GetVolumeByID(ctx, id)
+		if err != nil {
 			return nil, status.Errorf(
 				codes.NotFound,
 				"failed to find a volume with CSI ID %q: %v",
 				id,
 				err)
 		}
 
+		//nolint:gocritic // need to call .Destroy() for all volumes
+		defer vol.Destroy(ctx)
+		volumes[i] = vol
+	}
+
+	// extract the flatten mode
+	flattenMode, err := getFlattenMode(ctx, req.GetParameters())
+	if err != nil {
+		return nil, err
+	}
+	// Flatten the image if the flatten mode is set to FlattenModeForce
+	// before adding it to the volume group.
+	for _, vol := range volumes {
+		err = vol.HandleParentImageExistence(ctx, flattenMode)
+		if err != nil {
+			return nil, getGRPCError(err)
```

Comment:
consider adding more details to the error, include at least the volume name/id

Reply:
let me check that if HandleParentImageExistence already including it